grab.py

```diff
@@ -39,9 +39,20 @@ def get_data(self):
 
 
 def strip_tags(html):
+    """Function to strip HTML tags, but preserve <pre> and <code> tags for code snippets."""
     s = MLStripper()
     s.feed(html)
-    return s.get_data()
+
+    stripped_html = s.get_data()
+
+    # Handle <pre> and <code> tags manually for code snippets
+    stripped_html = stripped_html.replace('<pre>', '\n.. code-block:: python\n\n    ')
```

Comment:
It won't always be Python code. There are others using julia, c, ...

Also, wouldn't need to indent all the lines to the same column to work?